### DISCUSSION TITLE
Weact Studio 1.54" Epaper

### DISCUSSION THREAD

**Discussion by jbmfg** [peterhinch/micropython-nano-gui#76] [Q&A]
I have a Weact Studio 1.54" Epaper that im trying to get to work with an ESP32.  I was able to get it partially working with 
https://github.com/peterhinch/micropython-nano-gui/blob/master/drivers/epaper/epd29_ssd1680.py

I believe this module is actually a ssd1681 but the 1680 driver allows me to at least initialize and write to the screen.  However, when i do a fill(0) on it only 2/3 of the screen will go white.  fill(1) fills the entire screen.  The pixel command doesnt seem to work at all.  Is there any way to get this 1.54 screen working?  The only thing i tried was to change the resolution to 200x200 but that didnt see to have any effect.  Thanks in advance!

**Comment by peterhinch** [comment]
It's not possible for me to support hardware I don't possess. Writing a device driver inevitably involves a lot of experimentation. I suggest you find a driver from the manufacturer - which may be written in C - and see where it differs from mine. Another way forward is to find a datasheet for the SSD1681. These are the approaches I would use if I had a unit. Good luck!
